update/process/shardBlock.go

```diff
@@ -80,25 +80,35 @@ func NewShardBlockCreatorAfterHardFork(args ArgsNewShardBlockCreatorAfterHardFor
 	}, nil
 }
 
-// CreateNewBlock will create a new block after hardfork import
-func (s *shardBlockCreator) CreateNewBlock(
+// CreateBody will create a block body after hardfork import
+func (s *shardBlockCreator) CreateBody() (data.BodyHandler, []*update.MbInfo, error) {
+	mbsInfo, err := s.getPendingMbsAndTxsInCorrectOrder()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return s.CreatePostBody(mbsInfo)
+}
+
+// CreateBlock will create a block after hardfork import
+func (s *shardBlockCreator) CreateBlock(
+	bodyHandler data.BodyHandler,
 	chainID string,
 	round uint64,
 	nonce uint64,
 	epoch uint32,
-) (data.HeaderHandler, data.BodyHandler, error) {
-	if len(chainID) == 0 {
-		return nil, nil, update.ErrEmptyChainID
+) (data.HeaderHandler, error) {
+	blockBody, ok := bodyHandler.(*block.Body)
```

Comment:
Same her, we can avoid working with the body handler **or** add more useful functions on the body handler as to not require type assertions just about everywhere

Reply:
refactored